core/Archive/DataTableFactory.php

```diff
@@ -267,23 +312,26 @@ private function makeIndexedByRecordNameDataTable($blobRow)
      * @param array $keyMetadata The metadata to add to the table when it's created.
      * @return DataTable\Map
      */
-    private function createDataTableMapFromIndex($index, $resultIndices, $keyMetadata = array())
+    private function createDataTableMapFromIndex($index, $resultIndices, $keyMetadata)
     {
-        $resultIndexLabel = reset($resultIndices);
+        $result = new DataTable\Map();
+        $result->setKeyName(reset($resultIndices));
         $resultIndex = key($resultIndices);
 
         array_shift($resultIndices);
 
-        $result = new DataTable\Map();
-        $result->setKeyName($resultIndexLabel);
+        $hasIndices = !empty($resultIndices);
 
         foreach ($index as $label => $value) {
-            $keyMetadata[$resultIndex] = $label;
-
-            if (empty($resultIndices)) {
-                $newTable = $this->createDataTable($value, $keyMetadata);
-            } else {
+            if ($resultIndex === DataTableFactory::TABLE_METADATA_SITE_INDEX) {
+                $keyMetadata[$resultIndex] = new Site($label);
+            } elseif ($resultIndex === DataTableFactory::TABLE_METADATA_PERIOD_INDEX) {
+                $keyMetadata[$resultIndex] = $this->periods[$label];
+            }
```

Comment:
Small code clarity item: I think this might be clearer in a private method, ie, `$keyMetadata[$resultIndex] = $this->createTableIndexMetadata($resultIndex, $label);`